tests/ttnn/unit_tests/operations/test_moreh_clip_grad_norm.py

```diff
@@ -0,0 +1,224 @@
+            )
+
+            param = torch.nn.Parameter(torch.empty(input_shape, dtype=cpu_dtype))
+            grad = torch.empty(input_shape, dtype=cpu_dtype).uniform_(0, 2.5)
+            param.grad = grad
+
+            cpu_inputs.append(param)
+            npu_inputs.append(to_npu(grad.clone().bfloat16(), device, npu_dtype=npu_dtype))
+            input_shapes.append(input_shape)
+
+        cpu_total_norm = torch.nn.utils.clip_grad_norm_(cpu_inputs, max_norm, norm_type)
+        npu_total_norm = ttnn.operations.moreh.clip_grad_norm(npu_inputs, max_norm, norm_type)
+
+        expected_total_norm = cpu_total_norm
+        actual_total_norm = to_cpu(npu_total_norm, [1, 1, 1, 1])
+
+        rtol = atol = 0.1
+        # Check total_norm
+        pass_total_norm, out_total_norm = comp_allclose_and_pcc(
+            actual_total_norm.double(), expected_total_norm.double(), rtol=rtol, atol=atol
+        )
+        logger.debug(f"total_norm's {out_total_norm}")
+        assert pass_total_norm
+
+        # Check inputs
+        for i in range(num_parameters):
+            expected_input_i = cpu_inputs[i].grad.double()
+            actual_input_i = to_cpu(npu_inputs[i], input_shapes[i]).double()
+            pass_input_i, out_input_i = comp_allclose_and_pcc(expected_input_i, actual_input_i, rtol=rtol, atol=atol)
+            logger.debug(f"inputs[{i}]-shape[{input_shapes[i]}]'s {out_input_i}")
+            assert pass_input_i
+
+
+# @pytest.mark.skipif(is_wormhole_b0() or is_blackhole(), reason="Unsupported on WH and BH")
+# @pytest.mark.parametrize("error_if_nonfinite", [True, False])
+# def test_moreh_clip_grad_norm_with_error_if_nonfinite(error_if_nonfinite, device):
+#     torch.manual_seed(2023)
+
+#     cpu_dtype = torch.bfloat16
+#     npu_dtype = ttnn.bfloat16
+
+#     input_shape = [4, 4, 4 * TILE_HEIGHT, 4 * TILE_WIDTH]
+#     param = torch.nn.Parameter(torch.empty(input_shape, dtype=cpu_dtype))
+#     grad = torch.randn(input_shape, dtype=cpu_dtype)
+#     param.grad = grad
+
+#     max_norm = 1.0
+#     norm_type = float("nan")
+
+#     expected_error_msg = (
+#         f"The total norm of order {norm_type} for gradients from `parameters` is non-finite, so it cannot be clipped."
+#     )
+
+#     # Check vanilla torch behavior
+#     try:
+#         torch.nn.utils.clip_grad_norm_((param), max_norm, norm_type, error_if_nonfinite)
+#         assert not error_if_nonfinite
+#     except RuntimeError as actual_error_msg:
+#         assert expected_error_msg in str(actual_error_msg)
+#         assert error_if_nonfinite
+
+#     # Check tt behavior
+#     try:
+#         ttnn.operations.moreh.clip_grad_norm(
+#             [to_npu(param.grad.bfloat16(), device, npu_dtype=npu_dtype)], max_norm, norm_type, error_if_nonfinite
+#         )
+#         assert not error_if_nonfinite
+#     except RuntimeError as actual_error_msg:
+#         assert expected_error_msg in str(actual_error_msg)
+#         assert error_if_nonfinite
```

Comment:
Could you delete this before getting it approved?

Reply:
This test is included in the old test file so I copied it and change the deprecated moreh_clip_grad_norm operation to the new one to keep it consistent with the old version. Do you think I should delete it then?

Reply:
I’m not sure about the details, but since GitHub keeps a history, there’s usually no reason to keep unused code.
However, since this PR isn’t specifically for refactoring, it should be fine to leave it as is without deleting.